lib/assets/chart_builder/main.js

```diff
@@ -1,358 +1,306 @@
+        ctx.pushEvent("update_field", { field: "data_variable", value: this.fields.data_variable });
+      }
+    },
+
+    computed: {
+      axisOptions() {
+        const data = this.fields.data_variable;
+        return this.dataOptions[data];
+      },
+      noDataVariable() {
+        return !this.fields.data_variable;
+      },
+      noColorField() {
+        return !this.fields.color_field || this.fields.color_field === "__count__";
+      },
+      noYField() {
+        return !this.fields.y_field || this.fields.y_field === "__count__";
+      },
+      noXField() {
+        return !this.fields.x_field || this.fields.x_field === "__count__";
+      },
+    },
+
+    methods: {
+      handleFieldChange(event) {
+        const { name, value } = event.target;
+        ctx.pushEvent("update_field", { field: name, value });
+      },
+    },
+
+    components: {
+      BaseInput: {
+        props: {
+          label: {
+            type: String,
+            default: ''
+          },
+          modelValue: {
+            type: [String, Number],
+            default: ''
+          },
+        },
+        template: `
+          <div class="field">
+            <label class="input-label">{{ label }}</label>
+            <input
+              :value="modelValue"
+              @input="$emit('update:modelValue', $event.target.value)"
+              v-bind="$attrs"
+              class="input"
+            >
+          </div>
+        `
+      },
+      BaseSelect: {
+        props: {
+          label: {
+            type: String,
+            default: ''
+          },
+          modelValue: {
+            type: [String, Number],
+            default: ''
+          },
+          options: {
+            type: Array,
+            default: [],
+            required: true
+          },
+          required: {
+            type: Boolean,
+            default: false
+          },
+          count: {
+            type: Boolean,
+            default: false
+          },
+        },
+        methods: {
+          available(value, options) {
+            if (value === "__count__" || !value) {
+              return true;
+            } else {
+              return options.includes(value);
+            }
+          },
+        },
+        template: `
+          <div class="field">
+            <label class="input-label">{{ label }}</label>
+            <select
+              :value="modelValue"
+              v-bind="$attrs"
+              @change="$emit('update:modelValue', $event.target.value)"
+              class="input input--sm"
+              :class="{ unavailable: !available(modelValue, options) }"
+            >
+              <option v-if="!required"></option>
+              <option
+                v-for="option in options"
+                :value="option"
+                :key="option"
+                :selected="option === modelValue"
+              >{{ option }}</option>
+              <option v-if="count" value="__count__">COUNT(*)</option>
```

Comment:
We need to replicate `:selected` here too. One option to simplify would be to include `__count__` in `axisOptions()`. Then for the label we can either have mapping, or maybe just make the options a list of `[key, label]` altogether, whatever is simpler :)

Reply:
Actually we don't need to because this option is outside of v-for and will be auto-selected. :)
I quite like the solution we have today. We don't have to manipulate the data frame options and add something that doesn't really belong there.
We don't have to worry about conditionals to know which fields should or should not add the count. Today the field that wants to support the count just needs to make it explicit and we will do the rest!

Reply:
It feels a bit off to have a generic `BaseSelect` and then an axis-specific `__count__` option there. I was mostly wondering how we could minimize knowing about `__count__` everywhere.

> We don't have to worry about conditionals to know which fields should or should not add the count.

In the alternative version we wouldn't either, since we have `axisOptions()`, we can just include `__count__` there.

That said, it's up to you, I'm mostly throwing some ideas :)

Reply:
Why do you think it’s a bit off?
The repetition is not because count is not on the array, but because count behaves differently. So I need to check anyway.
Maybe we can use a const to storage it’s value…  How about?

Reply:
All I mean is that count is a special option specific to axis, so if we have it among the options passed to the component, then the component doesn't need to know about it.

> Maybe we can use a const to storage it’s value… How about?

Do you mean the label? That was my main question, if the label should be `{LABELS[option] || option}` (which is rather messy), or if rather the options should be `[key, label]` pairs, which I guess may not be that convenient here. So as said, I'm fine keeping it as is, we can always revisit if there are more special options :)